ocp_resources/csi_storage_capacity.py

```diff
@@ -0,0 +1,67 @@
+from ocp_resources.constants import TIMEOUT_4MINUTES
+from ocp_resources.resource import NamespacedResource
+
+
+class CSIStorageCapacity(NamespacedResource):
+    """
+    CSIStorageCapacity object. API reference:
+    https://docs.openshift.com/container-platform/4.12/rest_api/storage_apis/storage-apis-index.html#csistoragecapacity-storage-k8s-iov1
+    """
+
+    api_group = NamespacedResource.ApiGroup.STORAGE_K8S_IO
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        capacity=None,
+        maximum_volume_size=None,
+        node_topology=None,
+        storage_class_name=None,
+        teardown=True,
+        privileged_client=None,
+        yaml_file=None,
+        delete_timeout=TIMEOUT_4MINUTES,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the endpoints resource
+            namespace (str): Namespace of endpoints resource
+            client: (DynamicClient): DynamicClient for api calls
+            capacity (str): value reported by the csi driver
+            maximum_volume_size (str): maximum volume size reported by csi driver
+            storage_class_name (str): storage class name
+            node_topology (object): defines which node has access to the storage for which capacity was reported
+            teardown (bool): Indicates if the resource should be torn down at the end
+            privileged_client (DynamicClient): Privileged client for api calls
+            yaml_file (str): yaml file for the resource.
+            delete_timeout (int): timeout associated with delete action
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            privileged_client=privileged_client,
+            yaml_file=yaml_file,
+            delete_timeout=delete_timeout,
+            **kwargs,
+        )
+        self.capacity = capacity
+        self.node_topology = node_topology
+        self.storage_class_name = storage_class_name
+        self.maximum_volume_size = maximum_volume_size
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res.update(
+                {
+                    "maximumVolumeSize": self.maximum_volume_size,
+                    "nodeTopology": self.node_topology,
+                    "storageClassName": self.storage_class_name,
+                    "capacity": self.capacity,
+                }
```

Comment:
can be None?

Reply:
No a storage class name is a required field. I will add necessary check for all the resources.

Reply:
what about the other ones? (either check or add `optional` to docsting)

Reply:
I have added the check for only the required ones. But I think adding `optional` to the docstring is a good idea. I will do this for all the resources I am adding. Thank you!

Reply:
please also add to spec if they are actually passed (see my comment above)

Reply:
No spec for CSIStorageCapacity, but I understand what you meant. I will implement it.